### PR TITLE
[SwiftLanguageRuntime] Use mirrors to compute layout of objects.

### PR DESCRIPTION
**Pull request by dcci**
<rdar://problem/36477650>

### CODE REVIEW

**Comment by dcci** [comment]
This is a WIP, couple of tests are still failing, but the majority using resilience are now passing, which is good.

**Comment by dcci** [comment]
Mostly to understand what's the state of Linux/ELF.

**Comment by dcci** [comment]
Please test with following PR:
https://github.com/apple/swift/pull/20026

@swift-ci Please test

**Comment by dcci** [comment]
@swift-ci please test macos

**Comment by dcci** [comment]
@swift-ci please test

**Comment by dcci** [comment]
@swift-ci please test macos

File: source/Target/SwiftLanguageRuntime.cpp [hunk 1]
```diff
@@ -85,12 +85,19 @@ using namespace lldb_private;
 //----------------------------------------------------------------------
 // Destructor
 //----------------------------------------------------------------------
-SwiftLanguageRuntime::~SwiftLanguageRuntime() {}
+SwiftLanguageRuntime::~SwiftLanguageRuntime() {
+  delete reflection_ctx;
```
Comment (adrian-prantl) [inline source/Target/SwiftLanguageRuntime.cpp]: std::unique_ptr?

File: source/Target/SwiftLanguageRuntime.cpp [hunk 2]
```diff
@@ -103,14 +110,17 @@ void SwiftLanguageRuntime::SetupReflection() {
...
+    if (!obj_file)
+        return false;
     Address start_address = obj_file->GetHeaderAddress();
     auto load_ptr = static_cast<uintptr_t>(
         start_address.GetLoadAddress(&(m_process->GetTarget())));
     if (load_ptr == 0 || load_ptr == LLDB_INVALID_ADDRESS)
       return false;
-    reflection_ctx->addImage(swift::remote::RemoteAddress(load_ptr));
+    std::string module_path = module_sp->GetFileSpec().GetPath();
+    if (module_path.find("swift") != std::string::npos)
```
Comment (adrian-prantl) [inline source/Target/SwiftLanguageRuntime.cpp]: why does this work?

File: source/Target/SwiftLanguageRuntime.cpp [hunk 3]
```diff
@@ -184,8 +194,24 @@ void SwiftLanguageRuntime::SetupExclusivity() {
...
+        return true;
+    Address start_address = obj_file->GetHeaderAddress();
+    auto load_ptr = static_cast<uintptr_t>(
+        start_address.GetLoadAddress(&(m_process->GetTarget())));
+    if (load_ptr == 0 || load_ptr == LLDB_INVALID_ADDRESS)
+      return false;
+    if (!reflection_ctx)
+      return false;
+      std::string module_path = module_sp->GetFileSpec().GetPath();
+      if (module_path.find("swift") != std::string::npos)
```
Comment (adrian-prantl) [inline source/Target/SwiftLanguageRuntime.cpp]: this looks like it is the same code as above?

**Comment by dcci** [comment]
@swift-ci please test linux

**Comment by dcci** [comment]
Please test with following PR:
apple/swift#20026

@swift-ci Please test

**Comment by dcci** [comment]
Please test with following PR:
apple/swift#20026

@swift-ci Please test linux

File: source/Target/SwiftLanguageRuntime.cpp (hunk 1 above)
Comment (dcci) [reply source/Target/SwiftLanguageRuntime.cpp]: Fixed, thanks.

File: source/Target/SwiftLanguageRuntime.cpp (hunk 2 above)
Comment (dcci) [reply source/Target/SwiftLanguageRuntime.cpp]: That was just an hack to see if it worked on the bots. I implemented the correct logic now.

**Comment by dcci** [comment]
Please test with following PR:
apple/swift#20026

@swift-ci Please test linux

**Comment by dcci** [comment]
Please test with following PR:
apple/swift#20026

@swift-ci Please test

**Pull request merged**